### DISCUSSION TITLE
UPC Codes Not Scanning on ipadOS 14.7.1

### DISCUSSION THREAD

**Discussion by unarsu** [mebjas/html5-qrcode#268] [Q&A]
First let me thank mebjas for a great library. 

My issue is that UPC codes aren't scanned on ipadOS. I've tried both the demo app, and I did an integration with our web app. While it works great for 2D barcodes like QR codes (and Aztec codes-- what we use), it doesn't seem to recognize basic 1D UPC codes. I've tried printed examples (like these: https://barcode.tec-it.com/en/UPCA?data=72527273070 and https://barcode.tec-it.com/en/UPCE?data=0123456) as well as barcodes from retail packages.  The onScanSuccess function never fires.

Could someone else try UPC codes? I might be able to arrange a bug bounty to have this fixed-- I'd have to check with my manager. I'd also be happy to verify against an older release, or to do debugging.

Thanks.

**Comment by ROBERT-MCDOWELL** [comment]
did you try with other devices/computer and cameras?

**Reply by unarsu** [reply]
I did, with my Dell laptop’s webcam. It didn’t work there either but maybe it’s because it’s hard for me to center the paper easily without a backward facing camera.

**Comment by ROBERT-MCDOWELL** [comment]
what I know is barcode must be scanned with a camera with good focus and parallel to the barcode, it will be indeed goo to add a red line to help the scan to be parallel.

**Reply by unarsu** [reply]
I was using the new default UI that shows a bounding box. I can try adding a red line and see if that helps.

**Comment by ROBERT-MCDOWELL** [comment]
ok, you can why not make a PR if ii can fix it.

**Reply by unarsu** [reply]
I wonder if a better approach might be to try this technique: https://web.archive.org/web/20200324210817/http://www.oberhofer.co/how-barcode-localization-works-in-quaggajs/

**Comment by ROBERT-MCDOWELL** [comment]
I think it's already implemented if I recall...
